clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/cache/client/EcsCloudWatchAlarmCacheClient.java

```diff
@@ -71,33 +72,19 @@ protected EcsMetricAlarm convert(CacheData cacheData) {
   }
 
   public List<EcsMetricAlarm> getMetricAlarms(
-      String serviceName, String accountName, String region) {
+      String serviceName, String accountName, String region, String ecsClusterName) {
     List<EcsMetricAlarm> metricAlarms = new LinkedList<>();
-    // we can filter more here.
 
-    Collection<EcsMetricAlarm> allMetricAlarms = getAll(accountName, region);
```

Comment:
Before All the alarms for an ECS account/region where fetched and iterated through to match the service. This is extremely costly. 

After the change the ECSCluster is added during the caching cycles to the cache key id for the ECS provider in the alarms. We retrieve the IDs with ECS account/region/EcsClusterName and then try to match the service.